pkg/controller/registry/reconciler/grpc.go

```diff
@@ -338,13 +339,30 @@ func isRegistryServiceStatusValid(source *grpcCatalogSourceDecorator) (bool, err
 }
 
 func (c *GrpcRegistryReconciler) ensurePod(logger *logrus.Entry, source grpcCatalogSourceDecorator, serviceAccount *corev1.ServiceAccount, overwrite bool) error {
-	// currentLivePods refers to the currently live instances of the catalog source
-	currentLivePods := c.currentPods(logger, source)
-	if len(currentLivePods) > 0 {
+	// currentPods refers to the current pod instances of the catalog source
+	currentPods := c.currentPods(logger, source)
+
+	tmpPods := currentPods[:0]
```

Comment:
nifty! didn't know about this one

Reply:
Actually, this reminds me there's a new kid on the block (`slices.DeleteFunc`). I'll switch to that as the intent is more clear.